common/beerocks/bwl/include/bwl/ap_wlan_hal.h

```diff
@@ -157,6 +157,23 @@ class ap_wlan_hal : public virtual base_wlan_hal {
     virtual bool sta_bss_steer(const std::string &mac, const std::string &bssid, int chan,
                                int disassoc_timer, int valid_int) = 0;
 
+    /*!
+     * Set wifi credentials or remove VAP.
+     *
+     * @param [in] ssid SSID of VAP.
+     * @param [in] pass Password for authorization.
+     * @param [in] auth Authentication type.
+     * @param [in] enc Encryption type.
+     * @param [in] is_backhaul Type of VAP - backhaul STA or fronthaul BSS.
+     * @param [in] teardown Remove VAP.
+     *
+     * @return true on success or false on error.
+     */
+
+    virtual bool set_wifi_credentials(const std::string &ssid, const std::string &pass,
+                                      const std::string &auth, const std::string &enc,
+                                      uint8_t is_backhaul, uint8_t teardown) = 0;
```

Comment:
As @arnout has said, `auth` and `enc` should be enums, and `is_backhauk` and `teardown` should be bitfield.

Reply:
Changed to bssinfo list.